jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/StateBasedNodeInstance.java

```diff
@@ -396,4 +400,30 @@ protected boolean checkProcessInstance(Activation activation) {
         }
         return true;
     }
+    
+    protected void mapDynamicOutputData(Map<String, Object> results) {
+        if (results != null && !results.isEmpty()) {
+            VariableScope variableScope = (VariableScope) ((ContextContainer) getProcessInstance().getProcess()).getDefaultContext( VariableScope.VARIABLE_SCOPE );
+            VariableScopeInstance variableScopeInstance = (VariableScopeInstance)(VariableScopeInstance)getProcessInstance().getContextInstance(VariableScope.VARIABLE_SCOPE);;
```

Comment:
`VariableScopeInstance` casting is done twice. There are also 2 semicolons in the end of the line.

Reply:
will remove duplicates

Reply:
fixed